### ISSUE TITLE
Mac Issues

### ISSUE THREAD

**Issue by jLynx** [portapack-mayhem/MayhemHub#43]
For some reason Intel Macs have issues sending data and Apple Silicon Macs just do not want to connect. It will connect long enough to get the moto output out and then instantly disconnect after that. There must be some sort of bug on the HackRF side. I have validated this on other online WebSerial API websites too

**Comment by pedroCX486** [comment]
Interestingly I seem to not have those issues on my MBP M1 Pro, could it be an issue with only certain clones or MacOS versions?

**Comment by u-foka** [comment]
Same here: it connects, but when trying to send files larger thatn ~256 bytes the pp freezes.

M1 Pro in a  2021 MBP, Sonoma 14.2.1, Chrome 121.0.6167.85, Mayhem: n_240128

**Comment by tomzhu1024** [comment]
Similar issue: it can connect, but when I try to update to the latest release, both the web app and pp freeze.

2021 MacBook Pro w/ M1 Pro, macOS: Sonoma 14.2.1, Chrome: 121.0.6167.184, Mayhem: n_240214

**Comment by loewal** [comment]
> Similar issue: it can connect, but when I try to update to the latest release, both the web app and pp freeze.
> 
> 2021 MacBook Pro w/ M1 Pro, macOS: Sonoma 14.2.1, Chrome: 121.0.6167.184, Mayhem: n_240214

Same here on Mac Mini M2 Sonoma

**Comment by e-minguez** [comment]
> Same here: it connects, but when trying to send files larger thatn ~256 bytes the pp freezes.
> 
> M1 Pro in a 2021 MBP, Sonoma 14.2.1, Chrome 121.0.6167.85, Mayhem: n_240128

Same MBP M2. Portapack frozen after trying to send a small 5KB file.

**Comment by EricBoily** [comment]
same but with intel mac. 2018 MBP16. brand new portapack. 
when I click on "update" this happend in the command windows and the connection is cut.
```
ch> mkdir /FIRMWARE
Error calling delete_file: 8 exists
ch> fclose
no open file
ch> fopen /FIRMWARE/mayhem_nightly_n_240227_OCI.ppfw.tar
ok
ch> fseek 0
ok
ch> fwb 100000
send 100000 bytes
```

**Comment by tony1tf** [comment]
Is anyone who understands Macs and PortaPack looking into this problem? What debugging can we do to help?

**Comment by gullradriel** [comment]
The issue is on hold at the moment.
Please wait for us to resume it, soon, when our schedule will allow it.

**Comment by tony1tf** [comment]
Hi gullradriel,
Thanks for replying. As I said in my post, is there anything us users can do to help debug the problem? I've started looking at the code, but it is a bit outside my abilities.
Tony

**Comment by fuomag9** [comment]
m1, live screen updates are VERY slow. Commands to control seem to work but have some seconds latency

**Comment by Jayseeargh** [comment]
Updating firmware via HackRF and M2 Mac still a problem, update screen hangs on HackRF.app, flash updater in Portapack Utilities returns Bad File

**Comment by FaceNick** [comment]
A few days ago I overwrote the latest Mayhem firmware v2.0.2 and installed the Nightly Release firmware on my PortaPack. Surprisingly this firmware upgrade installation process went through perfectly on both iMac and Window Platforms, as I tested it on both. and it was very simple as well. I encourage anyone to try it.

IMPORTANT: Regarding Mac Users, there are a few things I want to point out.
a. While doing this installation, I noticed that it was not possible to do the overwriting or upgrade 
Process with another firmware while the SD card had existing folders in it. ONLY when I removed the existing folders from the SD card this allowed me to proceed.

b. Using the correct USB cable is also crucial to this process, I personally use the USB cable that came with the PortaPack. Of course you can also use an aftermarket USB cable as long as its able to send and receive data.


Below in detail is exactly what I did - Step by Step:
1. Remove your SD card from your PortaPack, then slide it in your Mac or Windows, then Remove or Delete all folders that are in your SD card, Now take your SD card and put it back in your PortaPack.

2. Go to [https://hackrf.app/](https://hackrf.app/?fbclid=IwZXh0bgNhZW0CMTAAAR0RY24xyeXmn12rVzn8Y0NtPq-YxxdeMzbRI43HKjow3DU0xLaDUEIPl1M_aem_VOPhj-_S6T6UAvkRouDvaw) and connect your PortaPack to your PC with your USB cable (make sure your PortaPack is turned ON) Choose option: "Update to latest nightly release" Wait for the firmware to finish installing and when its done then disconnect from your PC.
Note: For newbies, This video basically shows you what this step looks like (go directly to 3:53 and stop at 4:50) [https://www.youtube.com/watch?v=wzP0zWi85SI](https://www.youtube.com/watch?v=wzP0zWi85SI&fbclid=IwZXh0bgNhZW0CMTAAAR06Uy9dyOtd_Bnmkmjnv9gQ_9GPSNWEJvgs5Fp7bshcDfwe4h76aQrnuYM_aem_UGF0kFTgalXVA_XTVoHKQg) 

3. Once again, Remove the SD card from your PortaPack, then slide it back in your Mac or Windows, you will notice now that there are only 3 or 4 (can't remember) "new" folders in your SD card. Now go to [https://github.com/portapack-mayhem/mayhem-firmware/releases](https://l.facebook.com/l.php?u=https%3A%2F%2Fgithub.com%2Fportapack-mayhem%2Fmayhem-firmware%2Freleases%3Ffbclid%3DIwZXh0bgNhZW0CMTAAAR18v4kNJyxey9So8DN4TlBwu_59nYU9PUxgleJKYHUclv-5uFJLzowEa0c_aem_Q4gRq97Lhl1gociDEcG1bw&h=AT2t_LPvT5M8nA86PG9blM1u6JT8LA5nkHFhN2w83ng_CSd7MH5DENpVuSwF2iRe-20qLcr-6sc0x5g-by-SlMvU60ffNCXHkbRSRy7gy8UlqO12BuJKMs966Og3wtcS7PV5FicokA&__tn__=-UK-R&c%5B0%5D=AT2gLKg-pZ298LdeeN0Dv2lq_SZ-OxhajtT3ilC6C59BGZgpGKDscNTIP6iQuYBuLRVTmvKHCFCc0vKYFtESWBA_5RfcTflc-Vm48nvEDgVDj8TqRjbbUipqhNVRVlxps-1L5uiaquYQap4Q3a-2Tjn1v90HUSIRPhTiNjnFGijadaikZJoyMSIt3jkG3yoYDag9nMxsZ1pbWnNed5WwJaXvWjMwrDY) here you will see a list of Nightly Release firmwares, CLICK on the most latest dated release( for example: Nightly Release - 2024-11-19) as the page opens, then, scroll down and you will see several available zip files. Click and download this zip file -> mayhem_nightly_n_241119_COPY_TO_SDCARD.zip

4. After you unzip it on your PC, you will see several folders, Now, Copy/Paste ALL the folders into your SD card.
Note: As the folders are pasting into your SD card, there will be a few Pop Up's asking you to either Replace or Merge folders, simply "everytime" accept to "replace" and /or "merge" each time you are asked.

5. Finally, when all new folders are installed in your SD card, remove the SD card from your PC and put it back in your PortaPack, and that all, have fun, Enjoy the Nightly release firmware with all its great features.

![IMG_1027](https://github.com/user-attachments/assets/4f8f1cdb-aec9-4c94-a5ae-04bca89bc425)
![IMG_1028](https://github.com/user-attachments/assets/be612faf-412d-473f-9bbe-61145e296f5e)

**Comment by drewzh** [comment]
Unfortunately I wasn't as lucky as you @FaceNick - try as I might, no-matter how empty I got the card, it always hangs as soon as I try the update. Luckily I have a Linux desktop in the form of the SteamDeck to perform the update.

**Comment by CyberRatingsTim** [comment]
TLDR is just do the copy to SDCARD method. 

i tried about 15 different iterations to get it to copy, seems to freeze on the file copy. makes a 0b file and stops.

**Comment by n1k0ism** [comment]
The firmware update freezes after about 10%
Everything else seems to work fine. Its a great web app!

Hackrf one 2014 (micro usb, good cable) + portapack H4M, with and without battery enabled. 
Mac M1, Tahoe 26.3, Opera, n_260414

Flashers for ESPHome and Meshtastic work in my environment without any issues.

**Comment by we-sell-bags** [comment]
ok guys, i reliably get a mac  M1 silicon. 14.7.7  to reliably update firmware.
The only way you will get this sorted is if you list out the  osx version. & if you are on battery or PSU.

I did some work with Ruckus a few years back and issues they had with macs, we tracked it down to why Macs have such extensive battery life., basically they "break "the standards where it is not clearly documented.

so for mac first ting is get on a PSU, turn off power saving, or it will collapse the connectives to save power.

most likely the signalling on the firmware side is not 100% to standard.

also is the absolutely stupid insane transfer speed and the 4096 block size limit.
even when the  SDR is in "USB DEVICE" mode the transfers can take several hours over USB3


also the screen refresh is broken on osx, on some things like  the  BLE & ADS-B. ,the screen does not update & text is missing 

for example 
on teh device these show up correctly, but on the  Chrome , it is missing or does not update, even if it only changes every few min.
most possibly it is due to the apps not having any sort of screen update semaphore to signal changes.

<img width="630" height="504" alt="Image" src="https://github.com/user-attachments/assets/c1bc2c31-e6c2-4748-92a5-f1dd5b8b642d" />

**Comment by Jarod1230** [comment]
Hey!

I get the following problem. No stable or nightly. Clicking on any of these two upper buttons does nothing.

<img width="1067" height="702" alt="Image" src="https://github.com/user-attachments/assets/a35c4a99-f90e-4e7e-a4d4-947cb674e8a6" />

**Comment by guineau** [comment]
MacBook Pro M5 running macOS 27 Beta 4.

Using Chromium browser to connect to new PortaPack M4H Mayhem Pro. Connected fine. I was able to control the portapack UI from Mayhem Hub. Tried to install "latest nightly build" and it started downloading to portapack but eventually stalled. 

<img width="1107" height="948" alt="Image" src="https://github.com/user-attachments/assets/4d5d48b6-ae3b-42be-86fc-8d0034e81464" />

I tried refreshing the Chromium page but now Mayhem Hub will not load - it is stuck on the "Loading..." in the middle of the page. Closed browser tab.  Opened new tab. Same thing. Can't close Chromium because that is where I'm typing in this big report :) ).

Unplugged the Portapack and power-cycled it. Reloaded Chromian tab and Mayhem Hub came up and I'm attempting to flash again. 

Second time around it worked and I have the nightly on my portapack and all the apps!
